src/veh_interact.cpp

```diff
@@ -3186,6 +3186,7 @@ void veh_interact::complete_vehicle( player &p )
             // point because we don't want to put them back into the vehicle part
             // that just got removed).
             put_into_vehicle_or_drop( p, item_drop_reason::deliberate, resulting_items );
+            g->m.update_vehicle_cache( veh, veh->sm_pos.z );
```

Comment:
You can't access `veh` here anymore if the last part was removed. It was destroyed just a few lines above: line 3174 calls `map::destroy_vehicle`. You could set `veh` to null there and check for a valid pointer here before using it.